File: parent/3731 -- Find Missing Elements/mine1.py
class Solution:
    def findMissingElements(self, nums: List[int]) -> List[int]:
        # i prefer this
        # Time: O(n)
        # Space: O(n)

        hashN = set(nums)
        minim, maxim = min(nums), max(nums)
        res = []
        for i in range(minim, maxim):
            if i not in hashN:
                res.append(i)
        
        return res

File: parent/3731 -- Find Missing Elements/mine1.go
package main

func findMissingElements(nums []int) []int {
	minim, maxim := 100, 0
	hashSet := map[int]bool{}
	var res []int
	for _, x := range nums {
		minim = min(minim, x)
		maxim = max(maxim, x)
		hashSet[x] = true
	}
	for x := minim; x <= maxim; x++ {
		if _, exists := hashSet[x]; !exists {
			res = append(res, x)
		}
	}

	return res
}
